### PR TITLE
refactor voting

### PR DESCRIPTION
**Pull request by casey-chow**
I found it difficult to work around, so I needed to refactor the voting system first.

The main design improvements are as follows:

1. Separation of concerns between different stages of voting.
2. Separation of presentational aspects and logical aspects (i.e., api calls).

### CODE REVIEW

File: client/components/Voting/voting.tsx [hunk 1]
```diff
@@ -1,514 +1,26 @@
-import { Paginated } from '@feathersjs/feathers';
```
Comment (VinaiRachakonda) [inline client/components/Voting/voting.tsx:1]: This should be renamed to Voting.tsx to match with the Voting tag defined in pages/vote.tsx

Comment (casey-chow) [reply client/components/Voting/voting.tsx:1]: u right

File: client/components/Voting/VotingWorkflow.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    onChange={(evt) => this.setState({ prevote: evt.formData })}
+                  />
+                </>
+              ) : (
+                <>
+                  <Alert variant="success">
+                    You have already cast a prevote.
+                  </Alert>
+                  <VoteDisplay
+                    vote={vc.findVote(companyId, uc.user.id, VoteType.Prevote)}
```
Comment (VinaiRachakonda) [inline client/components/Voting/VotingWorkflow.tsx]: Shouldn't you do this after ComponentDidMount. I had problems with vote_id being undefined in the findVote method when clicking on the vote button (from pipeline page) on a company I submitted a vote on

File: client/components/Voting/VotingCompletion.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import VotingContainer, {
+  VotingStatus,
+} from '../../containers/VotingContainer';
+import { VoteType } from '../../schemas/vote';
+import { differenceWith } from 'lodash';
+import Table from 'react-bootstrap/lib/Table';
+import { PartnerVoteObj, archivedStates } from '../../schemas/company';
```
Comment (VinaiRachakonda) [inline client/components/Voting/VotingCompletion.tsx]: Remove unneeded imports

File: client/components/Voting/VotingCompletion.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+              <VotingCompletionTable
+                companyId={companyId}
+                votes={waitingOnPartners}
+                votingContainer={vc}
+              />
+            </>
+          ) : (
+            (vc.votingStatus(companyId, uc.user.id) !==
+              VotingStatus.AwaitingFinalization ||
+              uc.user.partner_position !== 'Managing Partner') && (
```
Comment (VinaiRachakonda) [inline client/components/Voting/VotingCompletion.tsx]: We have an enum for this right?

**Review by VinaiRachakonda — COMMENTED** [review]
You seem to be missing the 'overall_vote' field when rendering the final vote form.

File: client/components/Voting/VotingWorkflow.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      product_score: 1,
+      team_score: 1,
+      overall_vote: OverallVote.DontFund,
+    },
+    finalVote: {
+      fit_score: 1,
+      market_score: 1,
+      product_score: 1,
+      team_score: 1,
+      overall_vote: OverallVote.DontFund,
```
Comment (VinaiRachakonda) [inline client/components/Voting/VotingWorkflow.tsx:41]: Don't think you're rendering this?

**Review by VinaiRachakonda — CHANGES_REQUESTED** [review]

File: client/components/Voting/VotingCompletion.tsx (hunk 4 above)
Comment (casey-chow) [reply client/components/Voting/VotingCompletion.tsx]: Fixed

**Review by VinaiRachakonda — APPROVED** [review]

**Pull request merged**